### PR TITLE
Enhanced RDoc for io.c

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treated:
- IO#open
- IO#sysopen
- #open
- IO#putc
- IO#puts
- #p
- Kernel#display

### CODE REVIEW

File: io.c [hunk 1]
```diff
@@ -8370,29 +8378,48 @@ io_puts_ary(VALUE ary, VALUE out, int recur)
 
 /*
  *  call-seq:
- *     ios.puts(obj, ...)    -> nil
+ *    puts(*objects) -> nil
+ *    puts           -> newline
```
Comment (jeremyevans) [inline io.c:8382]: `puts` without arguments returns `nil` (it prints a newline, but it does not return a newline).

File: io.c [hunk 2]
```diff
@@ -8490,11 +8517,31 @@ rb_p_result(int argc, const VALUE *argv)
...
+ *    p r                 # => 0..4
+ *    p [r, r, r]         # => [0..4, 0..4, 0..4]
+ *    p                   # => nil
+ *
+ *  Output:
+ *
+ *     0..4
+ *     [0..4, 0..4, 0..4]
+
+  directly writes _obj_.+inspect+ followed by a
```
Comment (jeremyevans) [inline io.c:8544]: This looks wrong, it appears to be missing the starting `*` to signify it is a comment (it's technically still a comment, so it compiles fine, but it needs to be fixed before it is committed).

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks good, only a couple minor changes requested.

File: io.c (hunk 1 above)
Comment (BurdetteLamar) [reply io.c]: Fixed.

File: io.c (hunk 2 above)
Comment (BurdetteLamar) [reply io.c]: All from 'directly' to the end is residue from the previous text, which should have been, and now is, deleted.

File: io.c [hunk 3]
```diff
@@ -8379,7 +8379,7 @@ io_puts_ary(VALUE ary, VALUE out, int recur)
 /*
  *  call-seq:
  *    puts(*objects) -> nil
- *    puts           -> newline
+ *    puts           -> nil
```
Comment (jeremyevans) [inline io.c:8382]: You could remove the link completely, since `*objects` in the line above implies there could be 0 arguments provided.

**Review by jeremyevans — APPROVED** [review]
Looks good.  I recommend one minor tweak to `puts` `call-seq`, but I will leave that up to you.

**Pull request merged**